### PR TITLE
add a flag for dakuten to Shiritori

### PR DESCRIPTION
**Pull request by Riolku**
Under this flag, the word きょうし accepts じ to start the next word.

Thanks to  @CBaetz for creating the dakuten table.

### CODE REVIEW

**Comment by Riolku** [comment]
This flag caused some interesting interactions with the other flags. I've made notes where useful to talk about issues that came up, and ordering of processing, I also tested it quite a bit locally. Let me know if you have any concerns!

File: bot/src/bot_settings.js [hunk 1]
```diff
@@ -14,6 +14,7 @@ function onShiritoriForeverEnabledChanged(treeNode, channelID, newSettingValidat
 
 function onShiritoriForeverLaxLongVowelsChanged() {}
 function onShiritoriForeverSmallLettersChanged() {}
+function onShiritoriForeverDakutenChanged() {}
```
Comment (mistval) [inline bot/src/bot_settings.js:17]: I think we can delete these, and just omit the property from the settings.

File: bot/src/bot_settings.js [hunk 2]
```diff
@@ -328,6 +329,17 @@ module.exports = [
         convertInternalValueToUserFacingValue: SettingsConverters.createBooleanToStringConverter('Enabled', 'Disabled'),
         validateInternalValue: SettingsValidators.isBoolean,
       },
+      {
+        userFacingName: 'Dakuten',
```
Comment (mistval) [inline bot/src/bot_settings.js:333]: Maybe "Lax dakuten" would be a more descriptive setting name?

File: node-common/shiritori/japanese_game_strategy.js [hunk 3]
```diff
@@ -24,6 +25,8 @@ const largeHiraganaForSmallHiragana = {
 };
 
 function getNextWordMustStartWith(settings, currentWordReading) {
+  console.log(settings, currentWordReading);
```
Comment (mistval) [inline node-common/shiritori/japanese_game_strategy.js:28]: Let's remove the log statements (there's another one a little ways down too)

File: node-common/shiritori/japanese_game_strategy.js [hunk 4]
```diff
@@ -33,17 +36,19 @@ function getNextWordMustStartWith(settings, currentWordReading) {
   }
 
   if(settings.laxLongVowels && lengthenerForChar[secondLastCharacter] == finalCharacter) {
-    dupSettings = { ...settings };
+    let dupSettings = { ...settings };
     dupSettings.laxLongVowels = false;  // Don't use lax long vowels twice.
 
     // Get the normal "next word must start with" for both the last and second-last character
-    return getNextWordMustStartWith(dupSettings, currentWordReading).concat(
+    a = getNextWordMustStartWith(dupSettings, currentWordReading);
```
Comment (mistval) [inline node-common/shiritori/japanese_game_strategy.js:43]: Could we give this a more descriptive name than `a`?

**Review by mistval — CHANGES_REQUESTED** [review]
It seems good in my testing 👍

**Comment by Riolku** [comment]
So sorry for pushing code with some debug stuff leftover... Updated now.

**Comment by Riolku** [comment]
Sort of related, this function would really benefit from unit testing (though I'm much too lazy to set that up)...

**Comment by mistval** [comment]
> Sort of related, this function would really benefit from unit testing (though I'm much too lazy to set that up...)

Me too 😉I will let you know when the changes are deployed (probably tonight for the website but I might not get around to the bot until tomorrow).

**Comment by mistval** [comment]
API and website have been deployed.

**Comment by mistval** [comment]
The bot has been deployed.

**Pull request merged**